shex/diseaseHistoryShape.shex

```diff
@@ -0,0 +1,89 @@
+PREFIX : <http://purl.org/ejp-rd/cde/v030/shex/>
+PREFIX obo: <http://purl.obolibrary.org/obo/> 
+PREFIX sio: <http://semanticscience.org/resource/>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+
+# Symptom onset:
+
+:identifierShape IRI {
+    a [sio:SIO_000115] ;
+    sio:SIO_000020 @:symptomOnsetRoleShape ;
+    sio:SIO_000300 xsd:string
+}
+
+:personShape IRI { 
+    a [sio:SIO_000498] ;
+    sio:SIO_000228 @:symptomOnsetRoleShape ;
+    sio:SIO:000008 @:symptomOnsetAttributeShape
+}
+
+:symptomOnsetRoleShape IRI {
+    a [obo:OBI_0000093] ;
+    a [sio:SIO_000016] ;
+    rdfs:label xsd:string ;
+    sio:SIO_000356 @:symptomOnsetProcessShape
+}
+
+:symptomOnsetProcessShape IRI {
+    a [sio:SIO_000006] ;
+    a [sio:SIO_001052] ;
+    rdfs:label xsd:string ;
+    sio:SIO_000229 @:symptomOnsetOutputShape
+}
+
+:symptomOnsetOutputShape IRI {
+    a [sio:SIO_000015] ;
+    a [obo:HP_0030674 obo:HP_0003577 obo:NCIT_C124294 obo:HP_0003674] ;
+    rdfs:label xsd:string ;
+    sio:SIO_000300 xsd:date ;
+    sio:SIO_000628 @:symptomOnsetAttributeShape
+}
+
+:symptomOnsetAttributeShape IRI {
+    a [sio:SIO_000614] ;
+    a [obo:NCIT_C124353] ;
+    rdfs:label xsd:string
+}
+
+# Diagnosis date:
+
+:IdentifierShape IRI {
```

Comment:
Is it possible to use the **Expressions with Choices** "|" (http://shex.io/shex-primer/index.html#choices) in the `:identifierShape` and `:personShape` above to include these two shapes? Otherwise, you may need "metadata" to say: 
_for validating "diagnosis data", please use the capital shape `:IdentifierShape`, and for validating "age onset" please use the lowercase one `:identifierShape`_ 
@rajaram5 I am not sure if that is more reasonable and wonder how do you think ：D

Reply:
Im not sure if i understood everything. By checking the code i made a mistake by calling them differently, my intention was to call `:identifierShape` for both of them, due to they are the same ID and Entity

Another option is to fuse them into one shape refering to both Role's shapes

Reply:
Okay,  I thought you may want to create two different shapes. Fusing them should be the way to go. I think shapes should be unique within one single file and shapes using the same IRI should be automatically merged by "machines". @rajaram5

Reply:
Maybe you can fuse them like this https://github.com/ejp-rd-vp/CDE-semantic-model/blob/master/shex/sio-model/diseaseHistoryAndDiagnosisShape.shex#L21

Reply:
Like this
```
:identifierShape IRI {
    a [sio:SIO_000115] ;
    sio:SIO_000300 xsd:string;
    ( sio:SIO_000020 @:symptomOnsetRoleShape | sio:SIO_000020 @:diagnosisDateRoleShape)
}
```

@sxzhang1201  is this what you suggesting?

Reply:
Yes!